EventSource/EventSource.h

```diff
@@ -53,6 +53,19 @@ typedef void (^EventSourceEventHandler)(Event *event);
 /// @param timeoutInterval The request timeout interval in seconds. See <tt>NSURLRequest</tt> for more details. Default: 5 minutes.
 + (instancetype)eventSourceWithURL:(NSURL *)URL timeoutInterval:(NSTimeInterval)timeoutInterval;
 
+/// Returns a new instance of EventSource with the specified URL.
+///
+/// @param URL The URL of the EventSource.
+/// @param headers the HTTP Headers used in the request
++ (instancetype)eventSourceWithURL:(NSURL *)URL HTTPHeaders:(NSDictionary<NSString*, NSString *>*) headers;
```

Comment:
@neilco introducing nullability to the header will introduce warning unless all properties and parameters have nullability attributes